tardis/adapters/batchsystems/htcondor.py

```diff
@@ -4,51 +4,175 @@
+    )
+
+    try:
+        condor_status = await executor.run_command(cmd)
+    except CommandExecutionFailure as cef:
+        logger.warning(f"condor_status could not be executed due to {cef}!")
+        raise
+
+    return {
+        row["Machine"]: datetime.fromtimestamp(int(row["DaemonStartTime"]))
+        for row in csv_parser(
+            input_csv=condor_status.stdout,
+            fieldnames=tuple(class_ads.keys()),
+            delimiter="\t",
+        )
+        if row["Machine"] in htcondor_collectors
+    }
+
+
 async def htcondor_status_updater(
-    options: AttributeDict, attributes: AttributeDict, executor: Executor
+    options: AttributeDict,
+    attributes: AttributeDict,
+    executor: Executor,
+    cached_data: MappingProxyType,
 ) -> dict:
     """
     Helper function to call ``condor_status -af`` asynchronously and to translate
-    the output into a dictionary
+    the output into a dictionary.
+
+    If the HTCondor Collector has been running for less than 3600 seconds,
+    previously cached status data is used for machines that were already
+    available before the restart; otherwise, fresh status data is used.
 
     :param options: Additional options for the condor_status call. For example
         ``{'pool': 'htcondor.example'}`` will be translated into
         ``condor_status -af ... -pool htcondor.example``
-    :type options: AttributeDict
     :param attributes: Additional fields to add to output of the
         ``condor_status -af`` response.
-    :type attributes: AttributeDict
-    :return: Dictionary containing the output of the ``condor_status`` command
-    :rtype: dict
+    :param executor: Executor to run the ``condor_status`` command asynchronously.
+    :param ro_cached_data: Cached output from previous ``condor_status -af`` call
+    :return: Dictionary containing the processed output of the ``condor_status``
+        command, possibly merged with cached data depending on collector uptime.
     """
+    # copy options, since they are mutable
+    options = AttributeDict(**options)
 
-    attributes_string = f'-af:t {" ".join(attributes.values())}'
+    collector_start_dates = await htcondor_get_collector_start_dates(options, executor)
 
-    options_string = htcondor_cmd_option_formatter(options)
+    threshold = timedelta(seconds=3600)
+    now = datetime.now()
 
-    cmd = f"condor_status {attributes_string} -constraint PartitionableSlot=?=True"
-
-    if options_string:
-        cmd = f"{cmd} {options_string}"
+    earliest_start_date = min(collector_start_dates.values())
+    latest_start_date = max(collector_start_dates.values())
 
     htcondor_status = {}
 
+    if (now - earliest_start_date) < threshold:
+        # If all collectors have been running for less than 3600 seconds,
+        # use cached status for machines that were already available before the
+        # restart and update it with fresh data if available.
+        htcondor_status = {**cached_data}
```

Comment:
Maybe be explicit here.
```suggestion
        htcondor_status = cached_data.copy()
```